pkg/apis/v1alpha2/aksnodeclass.go

```diff
@@ -46,11 +46,109 @@ type AKSNodeClassSpec struct {
 	// Tags to be applied on Azure resources like instances.
 	// +optional
 	Tags map[string]string `json:"tags,omitempty"`
+	// Kubelet defines args to be used when configuring kubelet on provisioned nodes.
+	// They are a subset of the upstream types, recognizing not all options may be supported.
+	// Wherever possible, the types and names should reflect the upstream kubelet types.
+	// +kubebuilder:validation:XValidation:message="imageGCHighThresholdPercent must be greater than imageGCLowThresholdPercent",rule="has(self.imageGCHighThresholdPercent) && has(self.imageGCLowThresholdPercent) ?  self.imageGCHighThresholdPercent > self.imageGCLowThresholdPercent  : true"
+	// +optional
+	Kubelet *KubeletConfiguration `json:"kubelet,omitempty" hash:"ignore"`
+	// MaxPods is an override for the maximum number of pods that can run on a worker node instance.
+	// +kubebuilder:validation:Minimum:=0
+	// +optional
+	MaxPods *int32 `json:"maxPods,omitempty"`
 }
 
+// KubeletConfiguration defines args to be used when configuring kubelet on provisioned nodes.
+// They are a subset of the upstream types, recognizing not all options may be supported.
+// Wherever possible, the types and names should reflect the upstream kubelet types.
+// https://pkg.go.dev/k8s.io/kubelet/config/v1beta1#KubeletConfiguration
+// https://github.com/kubernetes/kubernetes/blob/9f82d81e55cafdedab619ea25cabf5d42736dacf/cmd/kubelet/app/options/options.go#L53
+//
+// AKS CustomKubeletConfig w/o CPUReserved,MemoryReserved,SeccompDefault
+// https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration?tabs=linux-node-pools
+type KubeletConfiguration struct {
```

Comment:
Just out of curiousity, where are all the fields for Kubelet Config we are supporting coming from?

K8s docs, AKS, Upstream, AWS refs?

Reply:
Primarily from AKS CustomKubeletConfig (as the comment says)

Reply:
Thx! Somehow I missed that.